modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/application/management/v1/ApplicationManagementSuccessTest.java

```diff
@@ -111,6 +94,43 @@ public void createApplication() throws Exception {
         assertNotBlank(createdAppId);
     }
 
+    @Test(dependsOnMethods = {"createApplication"})
```

Comment:
Let's leave the existing testGetAllApplications() method as it is to test the default application list api and introduce a new method to test the get application list api with required params (including the template id and version) in an extendable manner.